pkg/ingester/ingester_v2.go

```diff
@@ -809,21 +811,25 @@ func (i *Ingester) v2QueryStream(req *client.QueryRequest, stream client.Ingeste
 			ts.Samples = append(ts.Samples, client.Sample{Value: v, TimestampMs: t})
 		}
 		numSamples += len(ts.Samples)
-
-		timeseries = append(timeseries, ts)
 		numSeries++
-		batchSize++
-		if batchSize >= queryStreamBatchSize {
+		tsSize := ts.Size()
```

Comment:
How complicated would be adding a benchmark to the `v2QueryStream()`, to measure the performance impact of `ts.Size()`? I'm probably just paranoic, but if it's not much complicated we could measure it.